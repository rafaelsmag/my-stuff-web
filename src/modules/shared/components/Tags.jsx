import React from 'react'
import { arrayOf, shape, func, string } from 'prop-types'
import { Icon } from 'semantic-ui-react'
import { Div } from 'glamorous'
import { Metrics, Fonts, Colors } from '../../../config/constants'

const renderTags = (tags, onClick) => {
  return tags.map((tag) => (
    <Div css={styles.tagName} onClick={() => onClick(tag)}>{tag.name}</Div>
  ))
}

const Tags = ({ tags, onClickTag }) => (
  <Div css={styles.container}>
    <Div>
      <Icon name="tags" disabled size='small'/>
    </Div>
    {renderTags(tags, onClickTag)}
  </Div>
)

Tags.defaultProps = {
  onClickTag: () => {}
}

Tags.propTypes = {
  tags: arrayOf(shape({
    name: string.isRequired,
    _id: string
  })).isRequired,
  onClickTag: func
}

const styles = {
  container: {
    display: 'flex',
    flex: '1 1 auto',
    marginBottom: Metrics.xsSpacing,
    flexWrap: 'wrap'
  },
  tagName: {
    margin: `0 ${Metrics.xxsSpacing} ${Metrics.xxsSpacing} 0`,
    padding: Metrics.xxsSpacing,
    // backgroundColor: Colors.lightBackground,
    borderRadius: Fonts.size.tiny,
    fontSize: Fonts.size.tiny,
    lineHeight: Fonts.size.tiny,
    color: Colors.primary,
    // textTransform: 'uppercase',
    fontWeight: '600'
  }
}

export default Tags
