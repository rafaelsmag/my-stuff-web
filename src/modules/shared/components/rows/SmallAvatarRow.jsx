import React from 'react'
import { string } from 'prop-types'
import { Div } from 'glamorous'
import { Image } from 'semantic-ui-react'
import { Metrics, Fonts } from '../../../../config/constants'

const SmallAvatarRow = ({image, title}) => (
  <Div css={styles.container}>
    { image &&
      <Div css={styles.avatarImage}>
        <Image src={image} circular fluid />
      </Div>
    }
    <Div css={styles.title}>{title}</Div>
  </Div>
)

SmallAvatarRow.defaultProps = {
  image: undefined
}

SmallAvatarRow.propTypes = {
  title: string.isRequired,
  image: string
}

const styles = {
  container: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    marginTop: Metrics.xxsSpacing
  },
  avatarImage: {
    width: Metrics.tinyImage,
    marginRight: Metrics.tinySpacing
  },
  title: {
    textAlign: 'center',
    ...Fonts.style.detail
  }
}

export default SmallAvatarRow
