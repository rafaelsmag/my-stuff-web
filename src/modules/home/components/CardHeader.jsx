import React from 'react'
import { func, string, arrayOf, shape } from 'prop-types'
import { Div } from 'glamorous'
import { Metrics, Fonts } from '../../../config/constants'
import SmallAvatarRow from '../../shared/components/rows/SmallAvatarRow'
import Tags from '../../shared/components/Tags'

const CardHeader = ({title, tags, onClick, author}) => (
  <Div css={styles.cardHeader} onClick={onClick}>
    <Div css={styles.title}>{title}</Div>
    {
      tags &&
      <Tags tags={tags} />
    }
    {
      author &&
      <SmallAvatarRow title={author.name} image={author.image}/>
    }
  </Div>
)

CardHeader.defaultProps = {
  onClick: () => {},
  title: '',
  tags: undefined,
  author: undefined
}

CardHeader.propTypes = {
  onClick: func,
  title: string,
  tags: arrayOf(
    shape({
      name: string.isRequired,
      _id: string
    })
  ),
  author: shape({
    name: string,
    image: string
  })
}

export const styles = {
  cardHeader: {
    padding: Metrics.smallSpacing
    // borderBottom: `${Metrics.borderWidth} ${Colors.darkDivider} solid`
  },
  title: {
    paddingBottom: Metrics.xxsSpacing,
    paddingTop: Metrics.xxsSpacing,
    ...Fonts.style.h5
  }
}

export default CardHeader
