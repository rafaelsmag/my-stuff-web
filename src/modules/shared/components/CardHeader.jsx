import React from 'react'
import { func } from 'prop-types'
import { Div } from 'glamorous'
import { Metrics, Fonts, Colors } from '../../../config/constants'
import SmallAvatarRow from './rows/SmallAvatarRow'
import Tags from './Tags'

const CardHeader = ({title, tags, onClick, author}) => (
  <Div css={styles.cardHeader} onClick={onClick}>
    <Div css={styles.title}>{title}</Div>
    <Tags tags={tags} />
    {
      author &&
      <SmallAvatarRow title={author.name} image={author.image}/>
    }
  </Div>
)

CardHeader.defaultProps = {
  onClick: () => {},
  children: null
}

CardHeader.propTypes = {
  onClick: func
}

const styles = {
  cardHeader: {
    padding: Metrics.smallSpacing,
    borderBottom: `${Metrics.borderWidth} solid ${Colors.lightDivider}`
  },
  title: {
    marginBottom: Metrics.tinySpacing,
    ...Fonts.style.h5
  }
}

export default CardHeader
