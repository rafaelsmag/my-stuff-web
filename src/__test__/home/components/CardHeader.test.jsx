import React from 'react'
import { shallow } from 'enzyme'
import { Div } from 'glamorous'
import CardHeader, { styles } from '../../../modules/home/components/CardHeader'
import SmallAvatarRow from '../../../modules/shared/components/rows/SmallAvatarRow'
import Tags from '../../../modules/shared/components/Tags'

const props = {
  onClick: () => {},
  title: 'Lorem ipsum dolor sit amet',
  author: {
    image: 'https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png',
    name: 'Steve Sanders'
  },
  tags: [{_id: '1', name: 'Sobremesas'}, {_id: '2', name: 'Bolo'}]
}

describe('FeedCard Component', () => {
  const cardHeader = shallow(<CardHeader {...props} />)
  it('should redenders properly', () => {
    expect(cardHeader).toMatchSnapshot()
  })

  it('should not render the Tags component', () => {
    const { tags, ...propsWithoutTags } = props
    const wrapper = shallow(<CardHeader {...propsWithoutTags} />)
    expect(wrapper.find('Tags').exists()).toEqual(false)
  })

  it('should not render the SmallAvatarRow component', () => {
    const { author, ...propsWithoutAuthor } = props

    const wrapper = shallow(<CardHeader {...propsWithoutAuthor} />)
    expect(wrapper.find('SmallAvatarRow').exists()).toEqual(false)
  })
})
