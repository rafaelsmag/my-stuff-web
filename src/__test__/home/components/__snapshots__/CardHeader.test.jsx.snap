// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedCard Component should redenders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardHeader
    author={
        Object {
            "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
            "name": "Steve Sanders",
          }
    }
    onClick={[Function]}
    tags={
        Array [
            Object {
              "_id": "1",
              "name": "Sobremesas",
            },
            Object {
              "_id": "2",
              "name": "Bolo",
            },
          ]
    }
    title="Lorem ipsum dolor sit amet"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous.Div
          css={
                    Object {
                              "fontSize": "1.5em",
                              "paddingBottom": "4px",
                              "paddingTop": "4px",
                            }
          }
>
          Lorem ipsum dolor sit amet
</glamorous.Div>,
        <Tags
          onClickTag={[Function]}
          tags={
                    Array [
                              Object {
                                "_id": "1",
                                "name": "Sobremesas",
                              },
                              Object {
                                "_id": "2",
                                "name": "Bolo",
                              },
                            ]
          }
/>,
        <SmallAvatarRow
          image="https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png"
          title="Steve Sanders"
/>,
      ],
      "css": Object {
        "padding": "12px",
      },
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet",
          "css": Object {
            "fontSize": "1.5em",
            "paddingBottom": "4px",
            "paddingTop": "4px",
          },
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClickTag": [Function],
          "tags": Array [
            Object {
              "_id": "1",
              "name": "Sobremesas",
            },
            Object {
              "_id": "2",
              "name": "Bolo",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
          "title": "Steve Sanders",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous.Div
            css={
                        Object {
                                    "fontSize": "1.5em",
                                    "paddingBottom": "4px",
                                    "paddingTop": "4px",
                                  }
            }
>
            Lorem ipsum dolor sit amet
</glamorous.Div>,
          <Tags
            onClickTag={[Function]}
            tags={
                        Array [
                                    Object {
                                      "_id": "1",
                                      "name": "Sobremesas",
                                    },
                                    Object {
                                      "_id": "2",
                                      "name": "Bolo",
                                    },
                                  ]
            }
/>,
          <SmallAvatarRow
            image="https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png"
            title="Steve Sanders"
/>,
        ],
        "css": Object {
          "padding": "12px",
        },
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet",
            "css": Object {
              "fontSize": "1.5em",
              "paddingBottom": "4px",
              "paddingTop": "4px",
            },
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClickTag": [Function],
            "tags": Array [
              Object {
                "_id": "1",
                "name": "Sobremesas",
              },
              Object {
                "_id": "2",
                "name": "Bolo",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
            "title": "Steve Sanders",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
